### ISSUE TITLE
unittest: tearDown is not called if asyncSetUp fails

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#96139]
In `unittest.TestCase`, `tearDown()` is always called if `setUp()` is successful. It is only not called when `setUp()` fails. In `IsolatedAsyncioTestCase`, which is a subclass of `TestCase`, `tearDown()` is not called if `setUp()` is successful, but `asyncSetUp()` fails. I think is a problem.

### Linked PRs
* gh-99596

**Comment by sobolevn** [comment]
Yes, I think so too. I've proposed a solution to this in https://github.com/python/cpython/pull/99596
